### ISSUE TITLE
Disabled media in offer causes nullptr exception

### ISSUE THREAD

**Issue by sherlock1982** [otalk/rtcpeerconnection-shim#89]
How to reproduce:

1. Call setRemoteDescription with OFFER which contains enabled audio media and disabled video media.
2. Call setLocalDescription with generated ANSWER. 
3. Transceiver for video was not created and than you get nullptr exception here:

```
      var isIceLite = SDPUtils.matchPrefix(sessionpart,
          'a=ice-lite').length > 0;
      sections.forEach(function(mediaSection, sdpMLineIndex) {
        var transceiver = pc.transceivers[sdpMLineIndex];
        var iceGatherer = transceiver.iceGatherer;
  ^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^
```

My solution now is:

```
      var isIceLite = SDPUtils.matchPrefix(sessionpart,
          'a=ice-lite').length > 0;
      sections.forEach(function(mediaSection, sdpMLineIndex) {
        var transceiver = pc.transceivers[sdpMLineIndex];
        if (!transceiver)
              return;
        var iceGatherer = transceiver.iceGatherer;
```

Though of course I don't know all the details but this can give you some ideas on how to properly fix this  case.

**Comment by fippo** [comment]
Thank you! This was a bit worse since the answer created simply ignored that m-line.

I have a tentative fix in #90 but i doubt you're sending the rejected m-line just for fun and never use it again thereafter, right?

**Comment by sherlock1982** [comment]
I don't remember exactly why client behaves like that (though I wrote it some years ago :-) 
I think it just wanted to add video media and than enable/disable it. Don't forget that you can't remove media in renegotiation.

BTW Chrome and Firefox are absolutely ok with that so Edge should also work.

**Comment by fippo** [comment]
have you tried edge's native RTCPeerConnection? Its renegotiation support is still a bit better (but it lacks in other areas)

**Comment by fippo** [comment]
should be fixed in version 1.2.9. Let me know if it still happens!
